pkg/api/defaults_test.go

```diff
@@ -249,6 +250,54 @@ func TestAssignDefaultAddonImages(t *testing.T) {
 			t.Errorf("expected setDefaults to set Image %s in addon %s, but got %s", expected, addon.Name, actual)
 		}
 	}
+
+	for _, addon := range addons {
+		addon.Containers[0].Image = "customTestImage"
+	}
+
+	// Image should not be overridden in create scenarios.
+	mockCS = getMockBaseContainerService("1.10.8")
+	mockCS.Properties.OrchestratorProfile.OrchestratorType = Kubernetes
+	mockCS.Properties.OrchestratorProfile.KubernetesConfig.Addons = addons
+	mockCS.SetPropertiesDefaults(false, false)
+	modifiedAddons = mockCS.Properties.OrchestratorProfile.KubernetesConfig.Addons
+
+	for _, addon := range modifiedAddons {
+		expected := "customTestImage"
+		actual := addon.Containers[0].Image
+		if actual != expected {
+			t.Errorf("expected setDefaults to set Image %s in addon %s, but got %s", expected, addon.Name, actual)
+		}
+	}
+
+	// Image should be overridden in update scenarios.
+	mockCS = getMockBaseContainerService("1.10.8")
+	mockCS.Properties.OrchestratorProfile.OrchestratorType = Kubernetes
+	mockCS.Properties.OrchestratorProfile.KubernetesConfig.Addons = addons
```

Comment:
Can we make the `addons` object here a singleton, like the return value from a func? I'm a little concerned that as we progress through these tests we're overriding this reference unintentionally. I assume there is meant to be a sort of immutable canonical "baseline default addons spec" that we want to validate against (in the case of create, we want to validate that we can overwrite it, in the case of upgrade, we want to validate that overwriting it has no effect)

Reply:
I'm not sure I understand your comment. https://github.com/Azure/aks-engine/pull/1385/files#diff-daf624fb49e7cf74e670db52d41e949aL214 is where addons is defined.

Are you talking about `mockCS.Properties.OrchestratorProfile.KubernetesConfig.Addons` or `addons`?